cmake/externals.cmake

```diff
@@ -383,6 +383,12 @@ function(drake_add_external PROJECT)
     drake_add_foreign_external(${PROJECT})
   endif()
 
+  if(SKIP_DRAKE_BUILD)
```

Comment:
in my first pull request I had this logic outside of this function, in the regular drake CMakeLists.txt.  It was much nicer in here, but I'll revert it.